### DISCUSSION TITLE
Welcome to OpenWeedLocator Discussions!

### DISCUSSION THREAD

**Discussion by geezacoleman** [geezacoleman/OpenWeedLocator#16] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with members of the OpenWeedLocator community. We hope that you:
  * Ask questions about the OWL.
  * Share how you've built an OWL or plan to use it on farm or in your garden
  * Share ideas on how it could be improved or features that could give the OWL new opportunities
  * Engage with other community members and users of the OWL
  * Welcome others and are open-minded. 

  To get started, comment below with an introduction of yourself and tell us about why you're here and what you want to do with the OWL!

**Comment by ahogan27** [comment]
Hi All,

Angus here. Just getting started on this project. Slowly getting the hardware together now. I work as a precision ag consultant and have been teaching myself some python, so very keen on helping this project and developing my programming skills at the same time.

**Comment by gurucurlew** [comment]
Great to see this being developed locally, keen to give it a try. Installing today

**Reply by geezacoleman** [reply]
Awesome to hear - let us know how it goes.

**Comment by jbatesmd** [comment]
Rusty here,  love the idea of this project and have started it. (printed out the 3-D parts). Looking forward to making it work.  I actually did a Kaggle type project in my AI course detecting weeds.  I have seen where someone was using lasers to zap the weeds anyone trying that approach?

**Reply by geezacoleman** [reply]
That's great! Glad to hear you're building one.

If you're interested in the deep learning side of it, check out the green-on-green branch. I'm in the process of building in support for Google Coral + .tflite files. Works ok on my laptop so far, but need to test it on the Pi to see true frame rates.

Haven't seen anyone using this to activate lasers, but have considered it! Someone on YouTube assembled a [laser grass cutting](https://www.youtube.com/watch?v=WTPHsouuGq4&ab_channel=rctestflight) device using cheap diode laser arrays from a digital laser projector. They're only a [few hundred dollars](https://www.ebay.com/itm/334250776126?hash=item4dd2e6163e:g:AFAAAOSwSShhmLSM&amdata=enc%3AAQAHAAAA4IoFZoGgCsD1pGM8fMms1G6h6v56Lksiw3gMOgK4CsCsJeBj%2FLuXI2DpTquvCXC%2B21ogl6urUXHrpi5XzUxAj%2BFV8E53EbLCBpJseOBsWCmnBpekXDRpGEC9sH1E5tqwX00NHj%2FtMXGk3Hec4toe4MY85Tql34VtnRKPWMG1cjyaIfoWwJyhCuGkXCoLwkkIONzaK3%2BaB6OoMhrHe80wBys%2B4zQbrbSE2YjVm2CG6wne8VpsAXSpAiVJqu3K%2F3zM7RaLIHC5drkuXYyxp45O7MvGvSbB5T4t%2BMRS8kuOLcMI%7Ctkp%3ABFBM1qy6o9Jh) and he integrates it with a [fairly low cost](https://www.stratusleds.com/module) controller too.

Combined, the laser could be toggled on/off without too much hassle from the OWL and it would end up as a spot-laser system for weed control between rows of crop, or in fallow. With the right optics and a few of these side by side you could make a very neat and relatively cheap laser weeding device to cover a relatively wide path. Targeting the laser has always been a difficult part, but if you overcome that by using a lot of low cost arrays, it would make it substantially cheaper and also reduce a lot of failure points.

Quite a few safety considerations of course - laser light at that intensity is very dangerous. 

The efficacy of those lasers on weeds is also quite unknown - but also worth looking into.

**Comment by david-j-webster** [comment]
Good Day, David here. looking to build a system for work. as there are no RPi 4 around i am looking at the RPi 5. been thinking that the pi 5 is faster and has two camera ports. Any thoughts on doing double cameras and double spray areas with one pi 5. also had a thought to use finer spray tips to make a finer shot of herbicide. looking at 6 to 7 units on a three boom system.

**Reply by geezacoleman** [reply]
That sounds like a great project. What do you mean by a 3 boom system? As in have three separate spray lines for different products?

I only received my RPi5 yesterday, so haven't had a chance yet to test out different configurations, but it's certainly on the list. Still working on moving the code from Picamera to Picamera2, but should be almost done. It also means we can start using the Raspberry Pi Global Shutter instead of the HQ camera for some higher quality images.

Running two cameras with one Pi would be quite nice - either to increase resolution over the same area or just be more efficient with the somewhat expensive Pi. My gut feeling is that even the more powerful Pi won't have enough processing power for running two cameras + algorithms concurrently at high enough framerates, but that can/will be tested.

**Reply by david-j-webster** [reply]
Any time line on your upgrades. been working with a pi 3 and a jetson nano but would like to start with the upgraded camera(Pi Global Shutter) and maybe the pi 5 or a bigger Jetson

**Reply by geezacoleman** [reply]
Current timeline for the OWL to fix these compatibility issues and support Pi 5, Global Shutter etc. is April/May. 

Jetson is certainly on the cards, but will likely be later this year, towards July/August at least for basic OWL operation.

In the mean time we're priotising Raspberry Pi reliability and performance, green on green with the Google Coral and also in cab control. 

If you'd like to try adapting the OWl software to the Jetson Nano that would be great - a basic version should be quite straightforward, but adding Green-On-Green with proper Tensor RT support for real time use is more challenging.

Let me know your thoughts on the priority list. Haven't had as much time to work on this these past 6 months finishing off this thesis unfortunately!

**Comment by david-j-webster** [comment]
Guy
Thanks for the quick response. I run a sprayer for an irrigation company, my boom has three 2-meter sections. I am looking at using OWL on this boom, i would like a more precise spray, down even as low as 10 degree.  this would allow for a better control on the water banks.  Here we can only spray with a permit closer that 30 meters to flowing water, and the farmers do not like the possible drift.  I really like the thought of green-on-green,  that is something that no one offers, or can offer their industrial customers.  I have the 8 relay boards and would love to try doubling the resolution and halving the spray angle. it will double the cost but if that what it takes..


Thank you
David Webster


________________________________
